servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/RetryingHttpRequesterFilter.java

```diff
@@ -133,7 +133,8 @@ public HttpExecutionStrategy requiredOffloads() {
          */
         private ContextAwareRetryingHttpClientFilter(final FilterableStreamingHttpClient delegate) {
             super(delegate);
-            this.executor = delegate.executionContext().executor();
+            this.executor = delegate.executionContext().executionStrategy().hasOffloads() ?
```

Comment:
Fair point and something I considered while fixing the broken mocking. I decided I wanted to think more about what `executor()` should return if there is no offloading configured, ie. if it should return the `ioExecutor()`. I concluded probably not, especially if the context had been constructed with a non-default `Executor`. I wondered if build time was a better option to ensure predictable behavior.

Reply:
We can start with an internal utility that takes `ExecutionContext` and returns an `Executor` to avoid duplication and later consider making it public (as a utility or a method on `ExecutionContext`) if we see a need for that.

Reply:
Thomas had a good suggestion for an API that I want to evolve on a bit before implement. The idea is to provide a method that takes an enum describing the type of offloading (something like the enum `HttpExecutionStrategies.HttpOffload`) and returns the appropriate Executor.